From 923ad2767ac722c7e4ec40d0f03ca01283a99b3b Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Wed, 25 Nov 2015 15:25:55 +0100
Subject: window: Don't lose position information

Before calling gdk_window_move_resize(), store the full configure
request, not just width and height.

Fixes firefox randomly losing position of its dropdown windows.

https://bugzilla.gnome.org/show_bug.cgi?id=758609
---
 gtk/gtkwindow.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index 309a1fe..047ceaf 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -6121,8 +6121,7 @@ gtk_window_show (GtkWidget *widget)
        * and gdk_window_resize() below, rather than
        * queuing it.
        */
-      info->last.configure_request.width = configure_request.width;
-      info->last.configure_request.height = configure_request.height;
+      info->last.configure_request = configure_request;
       
       /* and allocate the window - this is normally done
        * in move_resize in response to configure notify
-- 
cgit v0.11.2

